pkg/server/server.go

```diff
@@ -161,9 +167,50 @@ func listenAndServe(ctx context.Context, clients *clients.Clients, handler http.
 					AdmissionReviewVersions: []string{"v1", "v1beta1"},
 				},
 			},
+		}, &v1.MutatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "rancher.cattle.io",
+			},
+			Webhooks: []v1.MutatingWebhook{
+				{
+					Name: "rancherfleet.cattle.io",
+					ClientConfig: v1.WebhookClientConfig{
+						Service: &v1.ServiceReference{
+							Namespace: namespace,
+							Name:      "rancher-webhook",
+							Path:      &mutationPath,
+							Port:      &port,
+						},
+						CABundle: secret.Data[corev1.TLSCertKey],
+					},
+					Rules: []v1.RuleWithOperations{
+						{
+							Operations: []v1.OperationType{
+								v1.Create,
+							},
+							Rule: v1.Rule{
+								APIGroups:   []string{"management.cattle.io"},
+								APIVersions: []string{"v3"},
+								Resources:   []string{"fleetworkspaces"},
+								Scope:       &clusterScope,
+							},
+						},
+					},
+					FailurePolicy:           &failPolicyFail,
+					SideEffects:             &sideEffectClassNone,
```

Comment:
This does have side effects, so needs to be SideEffectClassNoneOnDryRun and then don't create anything on dryrun

Reply:
Dryrun has properly been handled in here. https://github.com/rancher/webhook/pull/14/commits/b57e47194500ef13a3037afcca76896467678836#diff-0c5685479c4cd51ebba475dde0ae84021c2f1f333619a544c5c0d6eddf75a558R44-R46

Reply:
Still need to update this field to `SideEffectClassNoneOnDryRun`

Reply:
Changed